documentation/typespec-rest-api-dev-process.md

```diff
@@ -86,8 +87,51 @@ Please first review recommended folder structure detailed in [this document](htt
 
 5. Now the project has been set up. You can modify the sample and develop your own APIs with TypeSpec
 
+### 4. Prepare and submit a Pull Request for reviewing
 
-### 4. Generate or Refresh SDK code from a TypeSpec project
+1. Create a branch in your local repository for your changes.
+
+2. Create or update the TypeSpec files for your service.
+
+3. Add or update 'examples' files for each operation of your OpenAPI file.
+
+   The [oav](https://github.com/Azure/oav) provides two ways to generate Swagger examples:
+
+   1. Generating basic examples and then manually modify the values. It will generate two examples for each operation: one contains minimal properties set, the other contains the maximal properties set. Since the auto-generated examples consist of random values for most types, you need replace them with meaningful values.
+
+   ```bash
+   oav generate-examples openapi.json
+   ```
+
+   2. (**Recommended**) Generating high quality examples from API Scenario test. Refer to [API Test section](getstarted/providerhub/step03-api-testing.md). It will validate the API quality and generate Swagger examples from live traffic in API Scenario test.
+
+   ```bash
+   oav run <scenario-file> --spec <spec-file> --generateExample
+   ```
+
+4. Add/update the `readme.md` file in either the 'resource-manager' or 'data-plane' folder to specify the version and location of the OpenAPI files. The `readme.md` is needed for both management-plane and data-plane services for REST API Docs generation. For management-plane services, the `readme.md` is also needed for SDK generation -- see [generating client with autorest](https://github.com/Azure/autorest/blob/main/docs/generate/readme.md#keeping-your-options-in-one-place-the-preferred-option). The `readme.md` may contain generation options for multiple languages, separated into high-level sections.
+
+   Example:[sample-readme](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/samplefiles/samplereadme.md)
+
+5. Ensure all generated files under `resource-manager` or `data-plane` have been added to PR.
```

Comment:
There should be steps somewhere to generate the emitted swagger files:
- sync with the target branch in the azure-rest-api-specs repo
  ```bash
     git fetch upstream/<target-branch>
     git pull upstream <target-branch>  
  ```
- in the root directory
  `npm install`
- in the project directory,
   `npx tsp compile .`